src/app/shopping-cart/shopping-cart.component.ts

```diff
@@ -12,47 +16,43 @@ import { ProductIdQuantity } from '../model/ProductIdQuantity';
 })
 export class ShoppingCartComponent implements OnInit {
 
-  private order: Order | undefined;
-
   orderedProducts: OrderedProduct[] = [];
 
-  constructor(private productService: ProductsService, private orderService: OrderService) {
+  products : ProductIdQuantity[] = [];
+
+  columnsToDisplay = ['category', 'productName', 'price', 'quantity'];
+
+  constructor(private location: Location, private store: Store<AppState>) {
   }
 
   ngOnInit(): void {
-    this.order = this.orderService.getOrder();
-    this.getProducts();
+    this.store.select(selectAllOrderProducts).pipe(take(1)).subscribe((response) => {
+      this.products = response;
```

Comment:
First of all, I don't understand why you assign the response to products and then do getProducts again, which takes them once again from the state. It's enough if you select once and assign to products.

Secondly, I'd actually rather you placed the selector into a variable and passed it in the form component with async pipe. This component looks like it's getting a bit cluttered.

Reply:
Here when I first take the list of products from the selector I get a list of products of type ProductIdQuantity[] because this is the type that an Order has to contain for the product list, but in the shopping cart I have to display all the details for the products in the shopping cart so I did that function - getProducts(). That function iterates through the product list taken from the selector and takes the product details for each product.

I know that it is not very beautiful what I did here, but I did not find a better solution.